iceoryx_posh/test/moduletests/test_popo_chunk_sender.cpp

```diff
@@ -0,0 +1,550 @@
+{
+    auto chunk = m_chunkSender.allocate(sizeof(DummySample));
+    EXPECT_FALSE(chunk.has_error());
+    EXPECT_THAT(m_memoryManager.getMemPoolInfo(0).m_usedChunks, Eq(1u));
+}
+
+TEST_F(ChunkSender_test, allocate_MultipleChunks)
+{
+    auto chunk1 = m_chunkSender.allocate(sizeof(DummySample));
+    auto chunk2 = m_chunkSender.allocate(sizeof(DummySample));
+
+    EXPECT_FALSE(chunk1.has_error());
+    EXPECT_FALSE(chunk2.has_error());
+    if (!chunk1.has_error() && !chunk2.has_error())
+    {
+        // must be different chunks
+        EXPECT_THAT(*chunk1, Ne(*chunk2));
+    }
+    EXPECT_THAT(m_memoryManager.getMemPoolInfo(0).m_usedChunks, Eq(2u));
+}
+
+TEST_F(ChunkSender_test, allocate_Overflow)
+{
+    std::vector<iox::mepoo::ChunkHeader*> chunks;
+
+    // allocate chunks until MAX_CHUNKS_ALLOCATE_PER_SENDER level
+    for (size_t i = 0; i < iox::MAX_CHUNKS_ALLOCATE_PER_SENDER; i++)
+    {
+        auto chunk = m_chunkSender.allocate(sizeof(DummySample));
+        if (!chunk.has_error())
+        {
+            chunks.push_back(*chunk);
+        }
+    }
+
+    for (size_t i = 0; i < iox::MAX_CHUNKS_ALLOCATE_PER_SENDER; i++)
+    {
+        EXPECT_THAT(chunks[i], Ne(nullptr));
+    }
+    EXPECT_THAT(m_memoryManager.getMemPoolInfo(0).m_usedChunks, Eq(iox::MAX_CHUNKS_ALLOCATE_PER_SENDER));
+
+    // Allocate one more sample for overflow
+    auto chunk = m_chunkSender.allocate(sizeof(DummySample));
+    EXPECT_TRUE(chunk.has_error());
+    EXPECT_THAT(chunk.get_error(), Eq(iox::popo::ChunkSenderError::TOO_MANY_CHUKS_ALLOCATED_IN_PARALLEL));
+    EXPECT_THAT(m_memoryManager.getMemPoolInfo(0).m_usedChunks, Eq(iox::MAX_CHUNKS_ALLOCATE_PER_SENDER));
+}
+
+TEST_F(ChunkSender_test, freeChunk)
+{
+    std::vector<iox::mepoo::ChunkHeader*> chunks;
+
+    // allocate chunks until MAX_CHUNKS_ALLOCATE_PER_SENDER level
+    for (size_t i = 0; i < iox::MAX_CHUNKS_ALLOCATE_PER_SENDER; i++)
+    {
+        auto chunk = m_chunkSender.allocate(sizeof(DummySample));
+        if (!chunk.has_error())
+        {
+            chunks.push_back(*chunk);
+        }
+    }
+
+    EXPECT_THAT(m_memoryManager.getMemPoolInfo(0).m_usedChunks, Eq(iox::MAX_CHUNKS_ALLOCATE_PER_SENDER));
+
+    // free them all
+    for (size_t i = 0; i < iox::MAX_CHUNKS_ALLOCATE_PER_SENDER; i++)
+    {
+        m_chunkSender.free(chunks[i]);
+    }
+
+    EXPECT_THAT(m_memoryManager.getMemPoolInfo(0).m_usedChunks, Eq(0u));
+}
+
+TEST_F(ChunkSender_test, freeInvalidChunk)
+{
+    auto chunk = m_chunkSender.allocate(sizeof(DummySample));
+    EXPECT_FALSE(chunk.has_error());
+    EXPECT_THAT(m_memoryManager.getMemPoolInfo(0).m_usedChunks, Eq(1u));
+
+    auto errorHandlerCalled{false};
+    auto errorHandlerGuard = iox::ErrorHandler::SetTemporaryErrorHandler([&errorHandlerCalled](
+        const iox::Error, const std::function<void()>, const iox::ErrorLevel) { errorHandlerCalled = true; });
+
+    iox::mepoo::ChunkHeader* myCrazyChunk = new iox::mepoo::ChunkHeader();
```

Comment:
just a nitpick, but unique_ptr could be used to be more idiomatic with modern C++